### ISSUE TITLE
Weird ipex_optimize training loop printing behavior

### ISSUE THREAD

**Issue by fredlarochelle** [intel/intel-extension-for-pytorch#378]
### Describe the issue

When I use the `ipex_optimize` function on the latest `xpu_master` branch, I encounter some unusual behavior in the training loop. It appears to me that the training loop is printing all the output from the previous iterations??

I managed to reproduce this issue based on the examples provided in the documentation. It doesn't seems to happen when using a resnet straight from PyTorch, soo I have added a pretty basic/random custom model. Here are the results:

With the `ipex_optimize` line commented out:
```
Files already downloaded and verified
Starting training!!!
Epoch [1/100], Loss: 4.2976, Maximum memory: 2990.49072265625 MB in 63.68143582344055s.
Epoch [2/100], Loss: 3.7876, Maximum memory: 2990.49072265625 MB in 60.202733278274536s.
Epoch [3/100], Loss: 3.5808, Maximum memory: 2990.49072265625 MB in 60.42131733894348s.
Epoch [4/100], Loss: 3.4510, Maximum memory: 2990.49072265625 MB in 60.288352251052856s.
Epoch [5/100], Loss: 3.3530, Maximum memory: 2990.49072265625 MB in 60.44800615310669s.
Epoch [6/100], Loss: 3.2696, Maximum memory: 2990.49072265625 MB in 60.51012396812439s.
Epoch [7/100], Loss: 3.1963, Maximum memory: 2990.49072265625 MB in 60.290602684020996s.
Epoch [8/100], Loss: 3.1329, Maximum memory: 2990.49072265625 MB in 60.36489152908325s.
Epoch [9/100], Loss: 3.0686, Maximum memory: 2990.49072265625 MB in 60.22627568244934s.
Epoch [10/100], Loss: 3.0113, Maximum memory: 2990.49072265625 MB in 60.439422607421875s.
...
```

With the `ipex_optimize` line:
```
Files already downloaded and verified
Can not find transformers in your environment, disable ipex transformer optimize
Starting training!!!
[/home/fred/miniconda3/envs/ipex_build_env_2/lib/python3.10/site-packages/intel_extension_for_pytorch/frontend.py:568](https://vscode-remote+ssh-002dremote-002bdev1.vscode-resource.vscode-cdn.net/home/fred/miniconda3/envs/ipex_build_env_2/lib/python3.10/site-packages/intel_extension_for_pytorch/frontend.py:568): UserWarning: To reduce device memory usage on XPU, optimization are done inplace, setting the inplace argument to True.
  warnings.warn(
[/home/fred/miniconda3/envs/ipex_build_env_2/lib/python3.10/site-packages/intel_extension_for_pytorch/frontend.py:575](https://vscode-remote+ssh-002dremote-002bdev1.vscode-resource.vscode-cdn.net/home/fred/miniconda3/envs/ipex_build_env_2/lib/python3.10/site-packages/intel_extension_for_pytorch/frontend.py:575): UserWarning: Weight Prepack and Sample Input are both disabled on XPU. The Onednn Layout is automatically applied.
  warnings.warn(
[/home/fred/miniconda3/envs/ipex_build_env_2/lib/python3.10/site-packages/intel_extension_for_pytorch/frontend.py:582](https://vscode-remote+ssh-002dremote-002bdev1.vscode-resource.vscode-cdn.net/home/fred/miniconda3/envs/ipex_build_env_2/lib/python3.10/site-packages/intel_extension_for_pytorch/frontend.py:582): UserWarning: For XPU, the optimize_lstm(replace lstm with ipex_lstm) is unsupported, so disable it
  warnings.warn(
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [2/100], Loss: 3.9963, Maximum memory: 5685.54052734375 MB in 58.9573073387146s.
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [2/100], Loss: 3.9963, Maximum memory: 5685.54052734375 MB in 58.9573073387146s.
Epoch [3/100], Loss: 3.7262, Maximum memory: 5685.54052734375 MB in 58.99350166320801s.
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [2/100], Loss: 3.9963, Maximum memory: 5685.54052734375 MB in 58.9573073387146s.
Epoch [3/100], Loss: 3.7262, Maximum memory: 5685.54052734375 MB in 58.99350166320801s.
Epoch [4/100], Loss: 3.5885, Maximum memory: 5685.54052734375 MB in 58.92863440513611s.
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [2/100], Loss: 3.9963, Maximum memory: 5685.54052734375 MB in 58.9573073387146s.
Epoch [3/100], Loss: 3.7262, Maximum memory: 5685.54052734375 MB in 58.99350166320801s.
Epoch [4/100], Loss: 3.5885, Maximum memory: 5685.54052734375 MB in 58.92863440513611s.
Epoch [5/100], Loss: 3.4872, Maximum memory: 5685.54052734375 MB in 58.987996339797974s.
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [2/100], Loss: 3.9963, Maximum memory: 5685.54052734375 MB in 58.9573073387146s.
Epoch [3/100], Loss: 3.7262, Maximum memory: 5685.54052734375 MB in 58.99350166320801s.
Epoch [4/100], Loss: 3.5885, Maximum memory: 5685.54052734375 MB in 58.92863440513611s.
Epoch [5/100], Loss: 3.4872, Maximum memory: 5685.54052734375 MB in 58.987996339797974s.
Epoch [6/100], Loss: 3.4052, Maximum memory: 5685.54052734375 MB in 58.94254159927368s.
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [2/100], Loss: 3.9963, Maximum memory: 5685.54052734375 MB in 58.9573073387146s.
Epoch [3/100], Loss: 3.7262, Maximum memory: 5685.54052734375 MB in 58.99350166320801s.
Epoch [4/100], Loss: 3.5885, Maximum memory: 5685.54052734375 MB in 58.92863440513611s.
Epoch [5/100], Loss: 3.4872, Maximum memory: 5685.54052734375 MB in 58.987996339797974s.
Epoch [6/100], Loss: 3.4052, Maximum memory: 5685.54052734375 MB in 58.94254159927368s.
Epoch [7/100], Loss: 3.3340, Maximum memory: 5685.54052734375 MB in 59.03280735015869s.
Epoch [1/100], Loss: 4.4605, Maximum memory: 5685.54052734375 MB in 61.90187335014343s.
Epoch [2/100], Loss: 3.9963, Maximum memory: 5685.54052734375 MB in 58.9573073387146s.
Epoch [3/100], Loss: 3.7262, Maximum memory: 5685.54052734375 MB in 58.99350166320801s.
Epoch [4/100], Loss: 3.5885, Maximum memory: 5685.54052734375 MB in 58.92863440513611s.
Epoch [5/100], Loss: 3.4872, Maximum memory: 5685.54052734375 MB in 58.987996339797974s.
Epoch [6/100], Loss: 3.4052, Maximum memory: 5685.54052734375 MB in 58.94254159927368s.
Epoch [7/100], Loss: 3.3340, Maximum memory: 5685.54052734375 MB in 59.03280735015869s.
Epoch [8/100], Loss: 3.2679, Maximum memory: 5685.54052734375 MB in 59.110628843307495s.
...
```
And here is the code:
```
import torch
import torchvision
import torch.nn as nn
import intel_extension_for_pytorch as ipex
import time

LR = 0.001
DOWNLOAD = True
DATA = 'datasets/cifar100'
NUM_EPOCHS = 100

# Transformation pipeline for data preprocessing
transform = torchvision.transforms.Compose([
    torchvision.transforms.Resize((224, 224)),
    torchvision.transforms.ToTensor(),
    torchvision.transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
])

# CIFAR-100 dataset
train_dataset = torchvision.datasets.CIFAR100(
    root=DATA,
    train=True,
    transform=transform,
    download=DOWNLOAD,
)

# Data loader
train_loader = torch.utils.data.DataLoader(
    dataset=train_dataset,
    batch_size=256,
    shuffle=True,
    num_workers=4,
    pin_memory=True,
)

# Model definition
class CustomModel(nn.Module):
    def __init__(self, num_classes):
        super(CustomModel, self).__init__()

        # Define the layers of your custom model
        self.dropout = nn.Dropout(0.1)
        self.conv1 = nn.Conv2d(3, 32, kernel_size=3, stride=1, padding=1)
        self.relu1 = nn.ReLU()
        self.pool1 = nn.MaxPool2d(kernel_size=2, stride=2)

        self.conv2 = nn.Conv2d(32, 64, kernel_size=3, stride=1, padding=1)
        self.relu2 = nn.ReLU()
        self.pool2 = nn.MaxPool2d(kernel_size=2, stride=2)

        self.flatten = nn.Flatten()
        self.fc1 = nn.Linear(64 * 56 *56, 128)
        self.relu3 = nn.ReLU()
        self.fc2 = nn.Linear(128, num_classes)

    def forward(self, x):
        x = self.dropout(x)
        x = self.conv1(x)
        x = self.relu1(x)
        x = self.pool1(x)

        x = self.conv2(x)
        x = self.relu2(x)
        x = self.pool2(x)

        x = self.flatten(x)
        x = self.fc1(x)
        x = self.relu3(x)
        x = self.fc2(x)

        return x

# Instantiate the custom model
num_classes = 100
model = CustomModel(num_classes)

# Move model to GPU if available
device = torch.device("xpu" if torch.xpu.is_available() else "cpu")
model = model.to(device)

# Loss function and optimizer
criterion = torch.nn.CrossEntropyLoss()
optimizer = torch.optim.SGD(model.parameters(), lr = LR, momentum=0.9)

# IPEX optimize
model, optimizer = ipex.optimize(model, optimizer=optimizer, dtype=torch.float32)

# Training loop
model.train()

print("Starting training!!!")

for epoch in range(NUM_EPOCHS):
    epoch_loss = 0.0
    time_begin = time.time()

    for batch_idx, (data, target) in enumerate(train_loader):
        optimizer.zero_grad()
        data = data.to(device)
        target = target.to(device)
        output = model(data)
        loss = criterion(output, target)
        loss.backward()
        optimizer.step()
        epoch_loss += loss.item()
    avg_epoch_loss = epoch_loss / (batch_idx + 1)
    memory_allocated = torch.xpu.max_memory_allocated() / 1024.0 / 1024.0  # Convert bytes to MB
    total_time = time.time() - time_begin

    print(f"Epoch [{epoch + 1}/{NUM_EPOCHS}], Loss: {avg_epoch_loss:.4f}, Maximum memory: {memory_allocated} MB in {total_time}s.")
```
Additionally, I have noticed that when using `ipex_optimize`, the converge of the loss is slightly slower compared to not using it. Furthermore, even without using `ipex_optimize`, IPEX itself converges approximately 40% slower than CUDA when comparing the same code on both a T4 (from Colab) and an A6000 GPU. I have the same observation for multiple models with different architectures. It might be better that I open a separate issue for this observation tho...

**Comment by jgong5** [comment]
This is really weird that the message is actually dumped from your model script, not from inside ipex. Are you able to use pdb to break on the debug print and see what happens? 

For the slow convergence, it seems a bug to me. Yes, please file a separate issue with reproducers. Thanks!

**Comment by fredlarochelle** [comment]
@jgong5 Sorry for the delay, I am pretty busy, I will try to get back to you when I have a moment over the next few days, probably Wednesday.
